### ISSUE TITLE
Cuisine Classification

### ISSUE THREAD

**Issue by sharvari382** [sharvari382/as#3]
Project Overview
This project predicts the type of cuisine a restaurant serves based on various features such as location, pricing, and other restaurant attributes. The model is built using Python and scikit-learn, employing a Random Forest Classifier to achieve high predictive accuracy.

Code:
`
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.metrics import accuracy_score, precision_score, recall_score, r2_score
from sklearn.ensemble import RandomForestClassifier
import joblib

# USER SETTINGS
DATA_PATH = "Dataset .csv"             # path to your CSV
TARGET_COL = "Cuisine"                            # target column
DROP_COLS = ['Restaurant ID', 'Restaurant Name', 'Address']  # columns to drop
MODEL_OUT = "cuisine_rf_model.pkl"                # model save name
PREVIEW_ROWS = 5                                  # how many rows to preview
SHOW_PREDICTIONS = 10                             # number of test predictions to display

print("Loading dataset...")
df = pd.read_csv(DATA_PATH)

# Show preview
print("\n--- Dataset Preview ---")
print(df.head(PREVIEW_ROWS))

# Convert column names to python-friendly
df.columns = [f"Column_{i}" if not str(col).isascii() else str(col) for i, col in enumerate(df.columns)]

# Drop unwanted columns
for col in DROP_COLS:
    if col in df.columns:
        df.drop(columns=[col], inplace=True)

# Split X and y
y = df[TARGET_COL]
X = df.drop(columns=[TARGET_COL])

# Identify categorical features
cat_features = X.select_dtypes(include=['object']).columns.tolist()
num_features = X.select_dtypes(exclude=['object']).columns.tolist()

preprocess = ColumnTransformer(
    transformers=[
        ('cat', OneHotEncoder(handle_unknown='ignore'), cat_features),
        ('num', 'passthrough', num_features)
    ])

# Pipeline with RandomForest
model = Pipeline(steps=[
    ('preprocess', preprocess),
    ('classifier', RandomForestClassifier(n_estimators=300, random_state=42))
])

# Train-test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=42)

print("\nTraining model...")
model.fit(X_train, y_train)

# Evaluate
y_pred = model.predict(X_test)
acc = accuracy_score(y_test, y_pred)
prec = precision_score(y_test, y_pred, average='weighted')
rec = recall_score(y_test, y_pred, average='weighted')
r2 = r2_score(pd.factorize(y_test)[0], pd.factorize(y_pred)[0])

print("\n--- Model Performance ---")
print(f"Accuracy : {acc:.4f}")
print(f"Precision: {prec:.4f}")
print(f"Recall   : {rec:.4f}")
print(f"R² Score : {r2:.4f}")

# Sample predictions
print(f"\n--- {SHOW_PREDICTIONS} Sample Predictions ---")
for i in range(min(SHOW_PREDICTIONS, len(X_test))):
    print(f"Actual: {y_test.iloc[i]} --> Predicted: {y_pred[i]}")

`
Dataset: 

[Dataset .csv](https://github.com/user-attachments/files/21799326/Dataset.csv)
